src/api/rest/decorators/with-reducers.js

```diff
@@ -0,0 +1,128 @@
+import * as Actions from 'api/rest/actions';
+import { normalize } from 'normalizr';
+import { Alert } from 'react-native';
+
+const displayError = error => {
+  Alert.alert('API Error', error);
+};
+
+export const withReducers = Provider => {
+  const client = new Provider();
+
+  return new Proxy(withReducers, {
+    get: (c, namespace) => {
+      return new Proxy(client[namespace], {
+        get: (endpoint, call) => (...args) => (dispatch, getState) => {
+          // Used as a key for state.pagination
+          const actionName = `${namespace}_${call}`
+            .replace(/([A-Z])/g, '_$1')
+            .toUpperCase();
+
+          const action = Actions[actionName];
+
+          if (!action) {
+            return displayError(
+              `Unknown action. Did you forget to define Actions.${actionName}?`
+            );
+          }
+
+          // Identify if we have our magical last argument
+          const declaredArgsNumber = endpoint[call].length;
+          const isMagicArgAvailable = args.length === declaredArgsNumber;
```

Comment:
I really didn't know how to name this extra last argument which should be implemented in all methods definition.. 
This is an argument that is always at the end, and is used for both internal & external stuff. AdditionalArg doesn't seem to describe it enough

Reply:
I think it would be easier if we considered that there could be more than one `magic` parameters, the more it is quite possible, given how it is currently used:

```js
const { loadMore = false } = magicArg; // then maybe `param2`, `param3`... and these are additional parameters
```